### PR TITLE
Fix: Allows events like lottieAnimationStart to fire when used in a Capacitor project

### LINKED ISSUE #368: App Crashes On Launch
# Bug report - I/System.out: Error adding plugin de.dustplanet.cordova.lottie.LottieSplashScreen.

## Expected output -
11/09 00:22:08: Launching 'app' on HUAWEI VOG-L29.
App restart successful without requiring a re-install.
$ adb shell am start -n "com.samsungAfrica.app/com.samsungAfrica.app.MainActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -D
Waiting for application to come online: com.samsungAfrica.app | com.samsungAfrica.app.test
Waiting for application to come online: com.samsungAfrica.app | com.samsungAfrica.app.test
Connected to process 28222 on device 'huawei-vog_l29-DUM0219308015864'.
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
D/ActivityThread: Attach thread to application
I/ActivityThread: Init compatible state: true
D/HwFrameworkSecurityPartsFactory: HwFrameworkSecurityPartsFactory in.
I/HwFrameworkSecurityPartsFactory: add HwFrameworkSecurityPartsFactory to memory.
E/BehaviorCollectManager: Fail to acquire dataAnalyzerService...
W/ActivityThread: Application com.samsungAfrica.app is waiting for the debugger on port 8100...
I/System.out: Sending WAIT chunk
Connecting to com.samsungAfrica.app
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
I/.samsungAfrica.ap: Late-enabling -Xcheck:jni
E/.samsungAfrica.ap: Unknown bits set in runtime_flags: 0x8000
I/.samsungAfrica.ap: Reinit property: dalvik.vm.checkjni= false
    ReInitProperties: persist.vm.debug.dumpapi= false
D/ActivityThread: Attach thread to application
I/ActivityThread: Init compatible state: true
D/HwFrameworkSecurityPartsFactory: HwFrameworkSecurityPartsFactory in.
I/HwFrameworkSecurityPartsFactory: add HwFrameworkSecurityPartsFactory to memory.
E/BehaviorCollectManager: Fail to acquire dataAnalyzerService...
W/ActivityThread: Application com.samsungAfrica.app is waiting for the debugger on port 8100...
I/System.out: Sending WAIT chunk
I/System.out: Debugger has connected
    waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
Connected to the target VM, address: 'localhost:8311', transport: 'socket'
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/System.out: waiting for debugger to settle...
I/chatty: uid=10205(com.samsungAfrica.app) identical 1 line
I/System.out: waiting for debugger to settle...
I/System.out: debugger has settled (1410)
I/HwApiCacheMangerEx: apicache path=/storage/emulated/0 state=mounted key=com.samsungAfrica.app#10205#256
    need clear apicache,because volumes changed,oldCnt=0 newCnt=1
I/HwApiCacheMangerEx: apicache path=/storage/emulated/0 state=mounted key=com.samsungAfrica.app#10205#0
I/HwApiCacheMangerEx: need clear apicache,because volumes changed,oldCnt=1 newCnt=1
E/AwareLog: AtomicFileUtils: readFileLines file not exist: android.util.AtomicFile@c57ea96
I/HwPartIawareFactory: add android.common.HwPartIawareFactoryImpl to memory.
I/AwareBitmapCacher: init processName:com.samsungAfrica.app pid=28222 uid=10205
W/ComponentDiscovery: Class com.google.firebase.dynamicloading.DynamicLoadingRegistrar is not an found.
I/FirebaseApp: Device unlocked: initializing all Firebase APIs for app [DEFAULT]
I/FirebaseInitProvider: FirebaseApp initialization successful
I/HwApiCacheMangerEx: apicache path=/storage/emulated/0 state=mounted key=com.samsungAfrica.app#10205#256
I/HwApiCacheMangerEx: need clear apicache,because volumes changed,oldCnt=1 newCnt=1
I/ActivityThread: finishPreloaded preloadStatus 0
V/ActivityThread: callActivityOnCreate
E/.samsungAfrica.ap: Invalid ID 0x00000000.
I/DecorView[]:  old windowMode:0 new windoMode:1
W/.samsungAfrica.ap: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
    Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
I/WebViewFactory: Loading com.google.android.webview version 95.0.4638.74 (code 463807434)
W/.samsungAfrica.ap: Accessing hidden method Landroid/os/Trace;->isTagEnabled(J)Z (greylist, reflection, allowed)
    Accessing hidden method Landroid/os/Trace;->traceBegin(JLjava/lang/String;)V (greylist, reflection, allowed)
    Accessing hidden method Landroid/os/Trace;->traceEnd(J)V (greylist, reflection, allowed)
    Accessing hidden method Landroid/os/Trace;->asyncTraceBegin(JLjava/lang/String;I)V (greylist, reflection, allowed)
    Accessing hidden method Landroid/os/Trace;->asyncTraceEnd(JLjava/lang/String;I)V (greylist, reflection, allowed)
I/cr_WVCFactoryProvider: Loaded version=95.0.4638.74 minSdkVersion=29 isBundle=true multiprocess=true packageId=3
E/.samsungAfrica.ap: Invalid ID 0x00000000.
    Invalid ID 0x00000000.
I/cr_LibraryLoader: Successfully loaded native library
I/cr_CachingUmaRecorder: Flushed 8 samples from 8 histograms.
E/chromium: [ERROR:network_service_instance_impl.cc(179)] Failed to grant sandbox access to network context data for /data/user/0/com.samsungAfrica.app/app_webview/Default with result 7: No such file or directory (2)
D/HwCustConnectivityManagerImpl: isBlockNetworkRequestByNonAis, INVALID_SUBSCRIPTION_ID
I/ConnectivityManager: requestNetwork and the calling app is: com.samsungAfrica.app
D/HwCustConnectivityManagerImpl: isBlockNetworkRequestByNonAis, INVALID_SUBSCRIPTION_ID
D/ActivityThread: add activity client record, r= ActivityRecord{b013b68 token=android.os.BinderProxy@8e6edea {com.samsungAfrica.app/com.samsungAfrica.app.MainActivity}} token= android.os.BinderProxy@8e6edea
D/Capacitor: Starting BridgeActivity
W/System.err: java.lang.ClassNotFoundException: de.dustplanet.cordova.lottie.LottieSplashScreen
W/System.err:     at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:454)
        at java.lang.Class.forName(Class.java:379)
        at org.apache.cordova.PluginManager.instantiatePlugin(PluginManager.java:489)
W/System.err:     at org.apache.cordova.PluginManager.getPlugin(PluginManager.java:169)
        at org.apache.cordova.PluginManager.startupPlugins(PluginManager.java:97)
        at org.apache.cordova.PluginManager.init(PluginManager.java:86)
        at com.getcapacitor.cordova.MockCordovaWebViewImpl.init(MockCordovaWebViewImpl.java:58)
W/System.err:     at com.getcapacitor.Bridge$Builder.create(Bridge.java:1276)
        at com.getcapacitor.BridgeActivity.load(BridgeActivity.java:72)
        at com.getcapacitor.BridgeActivity.onStart(BridgeActivity.java:110)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1447)
        at android.app.Activity.performStart(Activity.java:8144)
W/System.err:     at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3922)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:235)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:215)
W/System.err:     at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:187)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2462)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:219)
        at android.app.ActivityThread.main(ActivityThread.java:8393)
W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
W/System.err: Caused by: java.lang.ClassNotFoundException: Didn't find class "de.dustplanet.cordova.lottie.LottieSplashScreen" on path: DexPathList[[zip file "/data/app/com.samsungAfrica.app-DnIEK5ZVRfN4kYt9xssvYg==/base.apk"],nativeLibraryDirectories=[/data/app/com.samsungAfrica.app-DnIEK5ZVRfN4kYt9xssvYg==/lib/arm64, /system/lib64, /hw_product/lib64, /system/product/lib64, /prets/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:209)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
W/System.err:     at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
    	... 25 more
I/System.out: Error adding plugin de.dustplanet.cordova.lottie.LottieSplashScreen.
I/BlockMonitor: dispatchingThrewException In MainThread
D/AndroidRuntime: Shutting down VM
I/QarthLog: [PatchStore] createDisableExceptionQarthFile
    [PatchStore] create disable file for com.samsungAfrica.app uid is 10205
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.samsungAfrica.app, PID: 28222
    java.lang.NullPointerException: Attempt to invoke virtual method 'void org.apache.cordova.CordovaPlugin.privateInitialize(java.lang.String, org.apache.cordova.CordovaInterface, org.apache.cordova.CordovaWebView, org.apache.cordova.CordovaPreferences)' on a null object reference
        at org.apache.cordova.PluginManager.getPlugin(PluginManager.java:171)
        at org.apache.cordova.PluginManager.startupPlugins(PluginManager.java:97)
        at org.apache.cordova.PluginManager.init(PluginManager.java:86)
        at com.getcapacitor.cordova.MockCordovaWebViewImpl.init(MockCordovaWebViewImpl.java:58)
        at com.getcapacitor.Bridge$Builder.create(Bridge.java:1276)
        at com.getcapacitor.BridgeActivity.load(BridgeActivity.java:72)
        at com.getcapacitor.BridgeActivity.onStart(BridgeActivity.java:110)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1447)
        at android.app.Activity.performStart(Activity.java:8144)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3922)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:235)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:215)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:187)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2462)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:219)
        at android.app.ActivityThread.main(ActivityThread.java:8393)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
I/Process: Sending signal. PID: 28222 SIG: 9
Disconnected from the target VM, address: 'localhost:8311', transport: 'socket'

## Checklist

-   [ ] If there is a (potential) plugin conflict, I've identified the conflicting plugin
-   [ ] I have added a valid version output
-   [ ] I have attached necessary information like a screenshot, example project or videos

### PR DESCRIPTION
**Pull request by dtarnawsky**
This closes #368

### CODE REVIEW

**Comment by timbru31** [comment]
Interesting - calling the parent(?) view yields in a usable WebView engine? 👍

**Comment by dtarnawsky** [comment]
> Interesting - calling the parent(?) view yields in a usable WebView engine? 👍

Yes, and according to one of the engineers on the Capacitor side that like would also work in a Cordova project, so it could be simplified so that it isn't conditional. I didn't have to test in a Cordova project yet to make that simplification.

**Comment by dtarnawsky** [comment]
Updated to use the call:
`(webView.getView() as WebView).evaluateJavascript`
rather than:
`webView.engine.evaluateJavascript`

This allows the code to work on both Cordova and Capacitor projects and was tested with:
[Cordova Sample](https://github.com/dtarnawsky/cs-cordova-lottie-splashscreen)
and
[Capacitor Sample](https://github.com/dtarnawsky/cs-lottie-splash)

File: src/android/LottieSplashScreen.kt [hunk 1]
```diff
@@ -357,4 +358,4 @@ class LottieSplashScreen : CordovaPlugin() {
     companion object {
         private const val LOG_TAG = "LottieSplashScreen"
     }
-}
+}
```
Comment (timbru31) [inline src/android/LottieSplashScreen.kt:361]:
```suggestion
}

```

**Review by timbru31 — APPROVED** [review]

**Comment by timbru31** [comment]
Thanks again!
I'll plan to make two new releases soon (one with the fix, one with the upgrade to lottie-android's new major)

**Comment by timbru31** [comment]
@dtarnawsky Neat <3, thanks for the hint. Definitely going to watch that video!
I'm currently a bit overloaded with two jobs - but OSS maintenance made it to the higher ranks of my free time prioritization (as it nearly stalled). Hopefully I can push something out by the end of the week.

**Comment by GimpMaster** [comment]
Hello @timbru31  - Just curious if you are going to do a new release with this fix in?

**Pull request merged**